src/main/java/seedu/taskify/logic/commands/EditCommand.java

```diff
@@ -85,10 +87,15 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_TASK);
         }
 
-        if (taskToEdit.getStatus().equals(new Status(StatusType.EXPIRED))) {
+        if (taskToEdit.getStatus().status.equals(StatusType.EXPIRED)
+                && !editedTask.getStatus().status.equals(StatusType.EXPIRED)) {
             throw new CommandException(MESSAGE_CANNOT_EDIT_EXPIRED_STATUS);
         }
 
+        if (!editedTask.getDate().isValidDateForEditing()) {
+            throw new CommandException(MESSAGE_EDIT_DATE_TO_PAST_DATE);
+        }
```

Comment:
Following up from my comment on where `isValidDateForEditing` should be placed, this line shouldnt be an issue if `isValidDateForEditing` is a method in `EditCommand` instead. Otherwise, I'm just highlighting these 3 lines because of Law of Demeter violation